mutiny-core/src/node.rs

```diff
@@ -801,92 +839,135 @@ impl<S: MutinyStorage> Node<S> {
+                            )
+                            .await?;
+
+                        let lsp_invoice = match client
+                            .get_lsp_invoice(InvoiceRequest {
+                                bolt11: Some(invoice.to_string()),
+                                user_channel_id,
+                                fee_id: lsp_fee.id,
+                            })
+                            .await
+                        {
+                            Ok(lsp_invoice_str) => Bolt11Invoice::from_str(&lsp_invoice_str)?,
+                            Err(e) => {
+                                log_error!(self.logger, "Failed to get invoice from LSP: {e}");
+                                return Err(e);
+                            }
+                        };
 
-            Ok(lsp_invoice)
-        } else {
-            Ok(invoice)
+                        if invoice.network() != self.network {
+                            return Err(MutinyError::IncorrectNetwork(invoice.network()));
+                        }
+
+                        if lsp_invoice.payment_hash() != invoice.payment_hash()
+                            || lsp_invoice.recover_payee_pub_key() != client.get_lsp_pubkey()
+                        {
+                            return Err(MutinyError::InvoiceCreationFailed);
+                        }
+
+                        Ok(lsp_invoice)
+                    }
+                    AnyLsp::LspsFlow(client) => {
+                        if has_inbound_capacity {
+                            Ok(self
+                                .create_internal_invoice(
+                                    Some(amount_sat),
+                                    None,
+                                    labels,
+                                    route_hints,
+                                )
+                                .await?)
+                        } else {
+                            let lsp_invoice = match client
+                                .get_lsp_invoice(InvoiceRequest {
+                                    bolt11: None,
+                                    user_channel_id,
+                                    fee_id: lsp_fee.id,
+                                })
+                                .await
+                            {
+                                Ok(lsp_invoice_str) => Bolt11Invoice::from_str(&lsp_invoice_str)?,
+                                Err(e) => {
+                                    log_error!(self.logger, "Failed to get invoice from LSP: {e}");
+                                    return Err(e);
+                                }
+                            };
+                            Ok(lsp_invoice)
+                        }
+                    }
+                }
+            }
+            None => Ok(self
+                .create_internal_invoice(amount_sat, None, labels, route_hints)
+                .await?),
```

Comment:
Yeah I restructured it a bit because it was getting too confusing for me to leave it how it was.  Basically you can think of the flow like 

-- is there an LSP configured?
    --No, then create normal invoice
    --Yes,
         --- do shared setup that all LSPs need
         --- match on type of LSP and do custom logic each one needs